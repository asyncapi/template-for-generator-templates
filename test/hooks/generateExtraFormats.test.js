const { existsSync } = require('fs');
const path = require('path');
const Generator = require('@asyncapi/generator');
const dummySpecUrl = 'https://rawcdn.githack.com/asyncapi/generator/v1.0.1/test/docs/dummy.yml';
//you always want to generate to new directory to make sure test runs in clear environment
const outputDir = path.resolve('test/temp/generateExtraFormats', Math.random().toString(36).substring(7));

describe('generateExtraFormats()', () => {
  jest.setTimeout(30000);

  beforeAll(async() => {
    const generator = new Generator(path.resolve(__dirname, '../../'), outputDir, { 
      forceWrite: true,
      templateParams: {
        svg: 'true',
        png: 'true',
        pdf: 'true'
      }
    });
    await generator.generateFromURL(dummySpecUrl);
  });
  
  it('svg diagram file is generated', async () => {
    const svg = existsSync(path.join(outputDir, 'index.svg'));
    expect(svg).toBeTruthy();
  });

  it('png diagram file is generated', async () => {
    const png = existsSync(path.join(outputDir, 'index.png'));
    expect(png).toBeTruthy();
  });

  it('pdf diagram file is generated', async () => {
    const pdf = existsSync(path.join(outputDir, 'index.pdf'));
    expect(pdf).toBeTruthy();
  });
});
  