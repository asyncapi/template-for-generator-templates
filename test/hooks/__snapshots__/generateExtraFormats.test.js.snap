// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateExtraFormats() generates correct svg diagram 1`] = `"<svg id=\\"mermaid-0\\" width=\\"100%\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" height=\\"430\\" style=\\"max-width: 585.4765625px;\\" viewBox=\\"-55 -20 592.9453125 424\\" max-width=\\"600\\"><style>#mermaid-0{font-family:\\"trebuchet ms\\",verdana,arial,sans-serif;font-size:16px;fill:#333;}#mermaid-0 .error-icon{fill:#552222;}#mermaid-0 .error-text{fill:#552222;stroke:#552222;}#mermaid-0 .edge-thickness-normal{stroke-width:2px;}#mermaid-0 .edge-thickness-thick{stroke-width:3.5px;}#mermaid-0 .edge-pattern-solid{stroke-dasharray:0;}#mermaid-0 .edge-pattern-dashed{stroke-dasharray:3;}#mermaid-0 .edge-pattern-dotted{stroke-dasharray:2;}#mermaid-0 .marker{fill:#333333;}#mermaid-0 .marker.cross{stroke:#333333;}#mermaid-0 svg{font-family:\\"trebuchet ms\\",verdana,arial,sans-serif;font-size:16px;}#mermaid-0 g.classGroup text{fill:#9370DB;fill:#131300;stroke:none;font-family:\\"trebuchet ms\\",verdana,arial,sans-serif;font-size:10px;}#mermaid-0 g.classGroup text .title{font-weight:bolder;}#mermaid-0 .classTitle{font-weight:bolder;}#mermaid-0 .node rect,#mermaid-0 .node circle,#mermaid-0 .node ellipse,#mermaid-0 .node polygon,#mermaid-0 .node path{fill:#ECECFF;stroke:#9370DB;stroke-width:1px;}#mermaid-0 .divider{stroke:#9370DB;stroke:1;}#mermaid-0 g.clickable{cursor:pointer;}#mermaid-0 g.classGroup rect{fill:#ECECFF;stroke:#9370DB;}#mermaid-0 g.classGroup line{stroke:#9370DB;stroke-width:1;}#mermaid-0 .classLabel .box{stroke:none;stroke-width:0;fill:#ECECFF;opacity:0.5;}#mermaid-0 .classLabel .label{fill:#9370DB;font-size:10px;}#mermaid-0 .relation{stroke:#333333;stroke-width:1;fill:none;}#mermaid-0 .dashed-line{stroke-dasharray:3;}#mermaid-0 #compositionStart,#mermaid-0 .composition{fill:#333333 !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #compositionEnd,#mermaid-0 .composition{fill:#333333 !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #dependencyStart,#mermaid-0 .dependency{fill:#333333 !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #dependencyStart,#mermaid-0 .dependency{fill:#333333 !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #extensionStart,#mermaid-0 .extension{fill:#333333 !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #extensionEnd,#mermaid-0 .extension{fill:#333333 !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #aggregationStart,#mermaid-0 .aggregation{fill:#ECECFF !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 #aggregationEnd,#mermaid-0 .aggregation{fill:#ECECFF !important;stroke:#333333 !important;stroke-width:1;}#mermaid-0 .edgeTerminals{font-size:11px;}#mermaid-0:root{--mermaid-font-family:\\"trebuchet ms\\",verdana,arial,sans-serif;}#mermaid-0 class{fill:apa;}</style><g></g><defs><marker id=\\"extensionStart\\" class=\\"extension\\" refX=\\"0\\" refY=\\"7\\" markerWidth=\\"190\\" markerHeight=\\"240\\" orient=\\"auto\\"><path d=\\"M 1,7 L18,13 V 1 Z\\"></path></marker></defs><defs><marker id=\\"extensionEnd\\" refX=\\"19\\" refY=\\"7\\" markerWidth=\\"20\\" markerHeight=\\"28\\" orient=\\"auto\\"><path d=\\"M 1,1 V 13 L18,7 Z\\"></path></marker></defs><defs><marker id=\\"compositionStart\\" class=\\"extension\\" refX=\\"0\\" refY=\\"7\\" markerWidth=\\"190\\" markerHeight=\\"240\\" orient=\\"auto\\"><path d=\\"M 18,7 L9,13 L1,7 L9,1 Z\\"></path></marker></defs><defs><marker id=\\"compositionEnd\\" refX=\\"19\\" refY=\\"7\\" markerWidth=\\"20\\" markerHeight=\\"28\\" orient=\\"auto\\"><path d=\\"M 18,7 L9,13 L1,7 L9,1 Z\\"></path></marker></defs><defs><marker id=\\"aggregationStart\\" class=\\"extension\\" refX=\\"0\\" refY=\\"7\\" markerWidth=\\"190\\" markerHeight=\\"240\\" orient=\\"auto\\"><path d=\\"M 18,7 L9,13 L1,7 L9,1 Z\\"></path></marker></defs><defs><marker id=\\"aggregationEnd\\" refX=\\"19\\" refY=\\"7\\" markerWidth=\\"20\\" markerHeight=\\"28\\" orient=\\"auto\\"><path d=\\"M 18,7 L9,13 L1,7 L9,1 Z\\"></path></marker></defs><defs><marker id=\\"dependencyStart\\" class=\\"extension\\" refX=\\"0\\" refY=\\"7\\" markerWidth=\\"190\\" markerHeight=\\"240\\" orient=\\"auto\\"><path d=\\"M 5,7 L9,13 L1,7 L9,1 Z\\"></path></marker></defs><defs><marker id=\\"dependencyEnd\\" refX=\\"19\\" refY=\\"7\\" markerWidth=\\"20\\" markerHeight=\\"28\\" orient=\\"auto\\"><path d=\\"M 18,7 L9,13 L14,7 L9,1 Z\\"></path></marker></defs><g id=\\"classid-dummyCreated-0\\" class=\\"classGroup\\" transform=\\"translate(59.08984375,0 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"148.8125\\" height=\\"92.625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"38.7890625\\">dummyCreated</tspan></text><line x1=\\"0\\" y1=\\"21.5625\\" y2=\\"21.5625\\" x2=\\"148.8125\\"></line><text x=\\"5\\" y=\\"31.5625\\" fill=\\"white\\" class=\\"classText\\"><tspan x=\\"5\\">prop1 integer</tspan><tspan x=\\"5\\" dy=\\"10\\">prop2 string</tspan><tspan x=\\"5\\" dy=\\"10\\">sentAt string</tspan><tspan x=\\"5\\" dy=\\"10\\">dummyArrayWithObject array</tspan><tspan x=\\"5\\" dy=\\"10\\">dummyArrayWithArray array</tspan><tspan x=\\"5\\" dy=\\"10\\">dummyObject object</tspan></text><line x1=\\"0\\" y1=\\"88.125\\" y2=\\"88.125\\" x2=\\"148.8125\\"></line><text x=\\"5\\" y=\\"103.125\\" fill=\\"white\\" class=\\"classText\\"></text></g><g id=\\"classid-sentAt-1\\" class=\\"classGroup\\" transform=\\"translate(208.3515625,342.90625 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"40.234375\\" height=\\"42.5625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"5\\">sentAt</tspan></text><line x1=\\"0\\" y1=\\"21.5\\" y2=\\"21.5\\" x2=\\"40.234375\\"></line><text x=\\"5\\" y=\\"31.5\\" fill=\\"white\\" class=\\"classText\\"><tspan x=\\"5\\">string</tspan></text><line x1=\\"0\\" y1=\\"38.0625\\" y2=\\"38.0625\\" x2=\\"40.234375\\"></line><text x=\\"5\\" y=\\"53.0625\\" fill=\\"white\\" class=\\"classText\\"></text></g><g id=\\"classid-dummyArrayWithObject-2\\" class=\\"classGroup\\" transform=\\"translate(167.046875,142.625 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"122.84375\\" height=\\"42.625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"5\\">dummyArrayWithObject</tspan></text><line x1=\\"0\\" y1=\\"21.5625\\" y2=\\"21.5625\\" x2=\\"122.84375\\"></line><text x=\\"5\\" y=\\"31.5625\\" fill=\\"white\\" class=\\"classText\\"><tspan x=\\"5\\">array</tspan></text><line x1=\\"0\\" y1=\\"38.125\\" y2=\\"38.125\\" x2=\\"122.84375\\"></line><text x=\\"5\\" y=\\"53.125\\" fill=\\"white\\" class=\\"classText\\"></text></g><g id=\\"classid-dummyArrayWithArray-3\\" class=\\"classGroup\\" transform=\\"translate(0,148.40625 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"117.046875\\" height=\\"31.0625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"5\\">dummyArrayWithArray</tspan></text><line x1=\\"0\\" y1=\\"21.5625\\" y2=\\"21.5625\\" x2=\\"117.046875\\"></line><text x=\\"5\\" y=\\"31.5625\\" fill=\\"white\\" class=\\"classText\\"></text><line x1=\\"0\\" y1=\\"26.5625\\" y2=\\"26.5625\\" x2=\\"117.046875\\"></line><text x=\\"5\\" y=\\"41.5625\\" fill=\\"white\\" class=\\"classText\\"></text></g><g id=\\"classid-dummyObject-4\\" class=\\"classGroup\\" transform=\\"translate(314.4921875,235.25 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"189.75\\" height=\\"52.625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"62.078125\\">dummyObject</tspan></text><line x1=\\"0\\" y1=\\"21.5625\\" y2=\\"21.5625\\" x2=\\"189.75\\"></line><text x=\\"5\\" y=\\"31.5625\\" fill=\\"white\\" class=\\"classText\\"><tspan x=\\"5\\">dummyObjectProp1 string</tspan><tspan x=\\"5\\" dy=\\"10\\">dummyObjectProp2 [CIRCULAR] object</tspan></text><line x1=\\"0\\" y1=\\"48.125\\" y2=\\"48.125\\" x2=\\"189.75\\"></line><text x=\\"5\\" y=\\"63.125\\" fill=\\"white\\" class=\\"classText\\"></text></g><g id=\\"classid-dummyInfo-5\\" class=\\"classGroup\\" transform=\\"translate(192.4453125,235.25 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"72.046875\\" height=\\"52.625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"9.8828125\\">dummyInfo</tspan></text><line x1=\\"0\\" y1=\\"21.5625\\" y2=\\"21.5625\\" x2=\\"72.046875\\"></line><text x=\\"5\\" y=\\"31.5625\\" fill=\\"white\\" class=\\"classText\\"><tspan x=\\"5\\">prop1 string</tspan><tspan x=\\"5\\" dy=\\"10\\">sentAt string</tspan></text><line x1=\\"0\\" y1=\\"48.125\\" y2=\\"48.125\\" x2=\\"72.046875\\"></line><text x=\\"5\\" y=\\"63.125\\" fill=\\"white\\" class=\\"classText\\"></text></g><g id=\\"classid-dummyRecursiveObject-6\\" class=\\"classGroup\\" transform=\\"translate(308.2578125,337.875 )\\"><rect x=\\"0\\" y=\\"0\\" width=\\"202.21875\\" height=\\"52.625\\" class=\\" \\"></rect><text y=\\"15\\" x=\\"0\\"><tspan class=\\"title\\" x=\\"45.5859375\\">dummyRecursiveObject</tspan></text><line x1=\\"0\\" y1=\\"21.5625\\" y2=\\"21.5625\\" x2=\\"202.21875\\"></line><text x=\\"5\\" y=\\"31.5625\\" fill=\\"white\\" class=\\"classText\\"><tspan x=\\"5\\">dummyRecursiveProp1 [CIRCULAR] object</tspan><tspan x=\\"5\\" dy=\\"10\\">dummyRecursiveProp2 string</tspan></text><line x1=\\"0\\" y1=\\"48.125\\" y2=\\"48.125\\" x2=\\"202.21875\\"></line><text x=\\"5\\" y=\\"63.125\\" fill=\\"white\\" class=\\"classText\\"></text></g><path d=\\"M59.08984375,77.80341225223137L43.408203125,84.44034354352614C27.7265625,91.07727483482091,-3.63671875,104.35113741741047,-19.318359375,118.70681870870523C-35,133.0625,-35,148.5,-35,163.9375C-35,179.375,-35,194.8125,-35,211.08333333333334C-35,227.35416666666666,-35,244.45833333333334,-35,261.5625C-35,278.6666666666667,-35,295.7708333333333,5.55859375,312.22200466779344C46.1171875,328.67317600225357,127.234375,344.4713520045072,167.79296875,352.370440005634L208.3515625,360.26952800676077\\" id=\\"edge0\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M195.1742166958808,92.625L200.72330557990065,96.79166666666667C206.2723944639205,100.95833333333333,217.37057223196027,109.29166666666667,222.91966111598015,117.625C228.46875,125.95833333333333,228.46875,134.29166666666666,228.46875,138.45833333333334L228.46875,142.625\\" id=\\"edge1\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M84.8065772896582,92.625L80.42605399138183,96.79166666666667C76.04553069310546,100.95833333333333,67.28448409655273,109.29166666666667,62.903960798276366,118.58854166666667C58.5234375,127.88541666666667,58.5234375,138.14583333333334,58.5234375,143.27604166666666L58.5234375,148.40625\\" id=\\"edge2\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M207.90234375,65.54646768758053L241.47981770833334,74.22622307298377C275.0572916666667,82.90597845838703,342.2122395833333,100.2654892291935,375.7897135416667,116.66399461459675C409.3671875,133.0625,409.3671875,148.5,409.3671875,163.9375C409.3671875,179.375,409.3671875,194.8125,409.3671875,206.69791666666666C409.3671875,218.58333333333334,409.3671875,226.91666666666666,409.3671875,231.08333333333334L409.3671875,235.25\\" id=\\"edge3\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M228.46875,185.25L228.46875,189.41666666666666C228.46875,193.58333333333334,228.46875,201.91666666666666,228.46875,210.25C228.46875,218.58333333333334,228.46875,226.91666666666666,228.46875,231.08333333333334L228.46875,235.25\\" id=\\"edge4\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M228.46875,287.875L228.46875,292.0416666666667C228.46875,296.2083333333333,228.46875,304.5416666666667,228.46875,313.7135416666667C228.46875,322.8854166666667,228.46875,332.8958333333333,228.46875,337.9010416666667L228.46875,342.90625\\" id=\\"edge5\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M362.9857976172351,287.875L355.64115947269596,292.0416666666667C348.29652132815676,296.2083333333333,333.60724503907835,304.5416666666667,314.5406016862059,315.3583121895919C295.4739583333333,326.17495771251714,272.0299479166667,339.4749154250342,260.3079427083333,346.1248942812927L248.5859375,352.7748731375513\\" id=\\"edge6\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M404.2392946863581,287.875L403.4272768219651,292.0416666666667C402.61525895757205,296.2083333333333,400.99122322878605,304.5416666666667,400.99122322878605,312.875C400.99122322878605,321.2083333333333,402.61525895757205,329.5416666666667,403.4272768219651,333.7083333333333L404.2392946863581,337.875\\" id=\\"edge7\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path><path d=\\"M414.4950803136419,337.875L415.3070981780349,333.7083333333333C416.11911604242795,329.5416666666667,417.74315177121395,321.2083333333333,417.7431517712139,312.875C417.74315177121395,304.5416666666667,416.11911604242795,296.2083333333333,415.3070981780349,292.0416666666667L414.4950803136419,287.875\\" id=\\"edge8\\" class=\\"relation\\" marker-end=\\"url(#extensionEnd)\\"></path></svg>"`;
